acme.go

```diff
@@ -140,13 +148,13 @@ func Register(client *http.Client, config *Config) error {
 // which the client will have to choose from and perform, to complete authorization.
 //
 // If client argument is nil, DefaultClient will be used.
-func authorize(client *http.Client, config *Config, domain string) (ChallengeSet, error) {
+func authorize(client *http.Client, config *Config, domain string) (*Authorization, error) {
 	if client == nil {
 		client = http.DefaultClient
 	}
 	nonce, err := fetchNonce(client, config.Endpoint.AuthzURL)
 	if err != nil {
-		return ChallengeSet{}, err
+		return &Authorization{}, err
```

Comment:
always return `nil` instead of a pointer to an empty struct. if the caller receives an error, they won't look at the first returned value anyway.